UltraStar Play/Assets/Scenes/SongSelect/Search/SongSearchControl.cs

```diff
@@ -262,10 +262,11 @@ public List<SongMeta> GetFilteredSongMetas(List<SongMeta> songMetas)
         string searchText = GetRawSearchText() != "#"
             ? GetSearchText()
             : "";
-        List<SongMeta> filteredSongs = songMetas
-            .Where(songMeta => searchText.IsNullOrEmpty()
-                               || SongMetaMatchesSearchedProperties(songMeta, searchText))
-            .ToList();
+        string[] searchTexts = searchText.Split(new char[0], StringSplitOptions.RemoveEmptyEntries);
+        List<SongMeta> filteredSongs = songMetas;
+        for (string searchWord in searchTexts) {
```

Comment:
Thanks for the PR, I think it is a neat idea.

One suggestion: I would prefer a single Where-clause over this imperative filtering in a loop.

For example, consider something like this (not tested):

```
List<SongMeta> filteredSongs = songMetas
            .Where(songMeta => searchTexts.IsNullOrEmpty()
                               || searchTexts.AllMatch(searchText => SongMetaMatchesSearchedProperties(songMeta, searchText)))
            .ToList();
```

IMO this would make it more obvious that a song must match all search texts.